Source/DafnyCore/Verifier/BoogieGenerator.Decreases.cs

```diff
@@ -141,20 +145,28 @@ public partial class BoogieGenerator {
         };
 
         Bpl.Expr zero = null;
+        Expression dafnyZero = null;
         string zeroStr = null;
-        if (types0[k].IsNumericBased(Type.NumericPersuasion.Int)) {
+        if (dafny0[k].Type.NormalizeExpandKeepConstraints().IsNumericBased(Type.NumericPersuasion.Int)) {
           zero = Bpl.Expr.Literal(0);
+          dafnyZero = Expression.CreateIntLiteral(dafny0[k].tok, 0);
           zeroStr = "0";
-        } else if (types0[k].IsNumericBased(Type.NumericPersuasion.Real)) {
+        } else if (dafny0[k].Type.NormalizeExpandKeepConstraints().IsNumericBased(Type.NumericPersuasion.Real)) {
           zero = Bpl.Expr.Literal(BaseTypes.BigDec.ZERO);
+          dafnyZero = Expression.CreateRealLiteral(dafny0[k].tok, BigDec.ZERO);
           zeroStr = "0.0";
         }
         if (zero != null) {
           Bpl.Expr bounded = Bpl.Expr.Le(zero, ee1[k]);
+          Expression boundedDafny = Expression.CreateLess(dafnyZero, dafny1[k]);
```

Comment:
Shouldn't this be a less-or-equal?

```suggestion
          Expression boundedDafny = Expression.CreateAtMost(dafnyZero, dafny1[k]);
```

Reply:
Yes! Good catch.